### PR TITLE
Refactor Sqlalchemy queries to 2.0 style (Part 5)

### PR DESCRIPTION
**Pull request by phanikumv**
This is a continuation of the effort to refactor the queries to sqlalchemy 2.0

related: https://github.com/apache/airflow/issues/28723

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/cli/commands/dag_command.py [hunk 1]
```diff
@@ -302,7 +302,9 @@ def dag_next_execution(args) -> None:
         print("[INFO] Please be reminded this DAG is PAUSED now.", file=sys.stderr)
 
     with create_session() as session:
-        last_parsed_dag: DagModel = session.query(DagModel).filter(DagModel.dag_id == dag.dag_id).one()
+        last_parsed_dag: DagModel = session.scalars(
+            select(DagModel).where(DagModel.dag_id == dag.dag_id)
+        ).one()
```
Comment (uranusjr) [inline airflow/cli/commands/dag_command.py:307]:
```suggestion
        last_parsed_dag: DagModel = session.scalar(
            select(DagModel).where(DagModel.dag_id == dag.dag_id)
        )
```

**Review by uranusjr — APPROVED** [review]
I made some minor changes.

File: tests/cli/commands/test_task_command.py [hunk 2]
```diff
@@ -500,15 +500,14 @@ def test_mapped_task_render_with_template(self, dag_maker):
         assert 'echo "2022-01-01"' in output
         assert 'echo "2022-01-08"' in output
 
-    @mock.patch("sqlalchemy.orm.session.Session.query")
+    @mock.patch("airflow.cli.commands.task_command.select")
```
Comment (uranusjr) [inline tests/cli/commands/test_task_command.py:503]: I don’t think you should do this; in practice `select` never raises `NoResultsFound` when it is called, only when its result is submitted for execution (via `execute`, `scalars`, etc.).

Comment (phanikumv) [reply tests/cli/commands/test_task_command.py:503]: Replaced it with `` @mock.patch("airflow.cli.commands.task_command.Session.scalars")``

File: airflow/models/renderedtifields.py [hunk 3]
```diff
@@ -236,14 +232,15 @@ def _do_delete_old_records(
...
+        # breakpoint()
         stmt = (
             delete(cls)
             .where(
                 cls.dag_id == dag_id,
                 cls.task_id == task_id,
                 tuple_not_in_condition(
                     (cls.dag_id, cls.task_id, cls.run_id),
-                    session.query(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
+                    select(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
```
Comment (uranusjr) [inline airflow/models/renderedtifields.py:242]: Damn, `tuple_not_in_condition` and `tuple_in_condition` do not work well with the new style. Let’s leave this out, I’ll try to convert them in a separate PR.

File: airflow/models/renderedtifields.py [hunk 4]
```diff
@@ -236,14 +232,15 @@ def _do_delete_old_records(
         session: Session,
     ) -> None:
         # This query might deadlock occasionally and it should be retried if fails (see decorator)
+        # breakpoint()
```
Comment (uranusjr) [inline airflow/models/renderedtifields.py]: Accident?

File: airflow/models/renderedtifields.py (hunk 3 above)
Comment (phanikumv) [reply airflow/models/renderedtifields.py:242]: it's failing specifically for MSSQL, for rest all databases it is working fine. Will check today if we need to have specific logic for that

Comment (uranusjr) [reply airflow/models/renderedtifields.py:242]: I figured this out, will push to this branch

File: airflow/models/renderedtifields.py (hunk 4 above)
Comment (phanikumv) [reply airflow/models/renderedtifields.py]: Removed it

**Pull request merged**